sdk/typescript/src/rpc/websocket-client.ts

```diff
@@ -117,8 +117,11 @@ export class WebsocketClient {
 			this.options.callTimeout,
 		);
 
-		// If an input ID is provided, this is a reconnect and we need to use that ID instead:
-		this.#subscriptions.set(input.id || id, {
+		// If an input ID is provided, this is a reconnect and we need to delete the existing subscription
+		if (input.id !== undefined) {
+			this.#subscriptions.delete(input.id);
```

Comment:
I'm a little confused why this would fix a memory leak. The set below should immediately set over whatever was previously set here?

Reply:
Oh I see, we no longer preserve the id of the original subscription. I'll need to look through the code to see if this expected.

Reply:
Yeah, there is also an id field in the map's value object, which will be used as new input.id the next time it gets reconnected.

Reply:
I believe that this will break the `unsubscribe` logic, since the unsubscribe callback returned from the original method will no-longer be able to find the subscription by it's original ID.

There are a few ways this could be solved, but the easiest that comes to mind is that the subscription map could be updated to store both the map key, and the subscription id.  these would be the same initially, but the key would stay fixed, while the subscription id would be new on each reconnect.

Reply:
I believe that this will break the unsubscribe logic, since the unsubscribe callback returned from the original method will no-longer be able to find the subscription by it's original ID.

There are a few ways this could be solved, but the easiest that comes to mind is that the subscription map could be updated to store both the map key, and the subscription id. these would be the same initially, but the key would stay fixed, while the subscription id would be new on each reconnect.

Reply:
Thanks @hayes-mysten for the review. I pushed a new commit that adds an `initialId` to the `SubscriptionRequest` so that the map's key will be using the very first requested id all the time meanwhile the actual latestId will be stored. When unsubscribe, it will use the initialId to retrieve the Subscription and send the unsub request